### PR TITLE
[IMPROVE] validateParams to accept different validators per request method

### PR DESCRIPTION
**Pull request by KevLehman**

### CODE REVIEW

File: apps/meteor/app/api/server/api.d.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ type Options = (
 			twoFactorOptions?: ITwoFactorOptions;
 	  }
 ) & {
-	validateParams?: ValidateFunction;
+	validateParams?: ValidateFunction | { [key in Method]?: ValidateFunction };
```
Comment (KevLehman) [inline apps/meteor/app/api/server/api.d.ts:71]: @ggazzo , the idea is to have `validateParams: { GET: xxxx, POST: xxxx }` so individual handlers can have its own validator

File: apps/meteor/tests/end-to-end/api/livechat/10-departments.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
+/* eslint-env mocha */
+import { expect } from 'chai';
+import type { ILivechatDepartment } from '@rocket.chat/core-typings';
+import { Response } from 'supertest';
+
+import { getCredentials, api, request, credentials } from '../../../data/api-data.js';
+import { updatePermission, updateSetting } from '../../../data/permissions.helper';
+import { makeAgentAvailable, createAgent, createDepartment } from '../../../data/livechat/rooms.js';
+
+describe.only('LIVECHAT - Departments', function () {
```
Comment (KevLehman) [inline apps/meteor/tests/end-to-end/api/livechat/10-departments.ts:10]:
```suggestion
describe('LIVECHAT - Departments', function () {
```

File: packages/rest-typings/src/v1/omnichannel.ts [hunk 3]
```diff
@@ -381,7 +381,94 @@ const LivechatDepartmentSchema = {
...
+				},
+				remove: {
+					type: 'array',
+					items: {
+						type: 'string',
+					},
+				},
+			},
+			nullable: true,
+		},
```
Comment (KevLehman) [inline packages/rest-typings/src/v1/omnichannel.ts:465]:
```suggestion
		agents: {
			type: 'array',
			items: {
				type: 'object',
				properties: {
					agentId: {
						type: 'string',
					},
				},
				required: ['agentId'],
				additionalProperties: false,
			},
			nullable: true,
		},
```

File: apps/meteor/app/livechat/server/api/lib/departments.ts [hunk 4]
```diff
@@ -160,18 +157,13 @@ export async function findDepartmentAgents({
 }
 
 export async function findDepartmentsBetweenIds({
-	uid,
 	ids,
 	fields,
 }: {
 	uid: string;
```
Comment (murtaza98) [inline apps/meteor/app/livechat/server/api/lib/departments.ts:163]:
```suggestion
```

Comment (murtaza98) [reply apps/meteor/app/livechat/server/api/lib/departments.ts:163]: This param is no longer required here so we can delete it

File: packages/rest-typings/src/v1/omnichannel.ts [hunk 5]
```diff
@@ -381,7 +381,90 @@ const LivechatDepartmentSchema = {
...
+		agents: {
+			type: 'array',
+			items: {
+				type: 'object',
+				properties: {
+					agentId: {
+						type: 'string',
+					},
+				},
```
Comment (murtaza98) [inline packages/rest-typings/src/v1/omnichannel.ts:456]: This has a couple of more props namely `count` & `order`. Added them over here -> https://github.com/RocketChat/Rocket.Chat/pull/26410

**Review by murtaza98 — CHANGES_REQUESTED** [review]
@KevLehman FYI: Left some suggestions over here -> https://github.com/RocketChat/Rocket.Chat/pull/26410

**Review by ggazzo — APPROVED** [review]

**Pull request merged**